sqlx-macros-core/src/derives/attributes.rs

```diff
@@ -281,7 +281,7 @@ pub fn check_struct_attributes(
 
     assert_attribute!(
         !attributes.transparent,
-        "unexpected #[sqlx(transparent)]",
+        "unexpected #[sqlx(transparent)] - only valid for structs with exactly one field",
```

Comment:
```suggestion
        "#[sqlx(transparent)] is only valid for structs with exactly one field",
```

It's no longer unexpected.